lib/github.js

```diff
@@ -19,20 +24,33 @@ module.exports.updateRelease = async ({ formatted, tag, verbose }) => {
 
   const repo = GITHUB_REPO || _exec('basename `git rev-parse --show-toplevel`').toString().trim()
 
-  const { data: { name, id: release_id } } = await octokit.request('GET /repos/:owner/:repo/releases/tags/:tag', {
-    owner,
-    repo,
-    tag,
-  })
-  log(`Release ${name} (ID: ${release_id}) found based on tag: ${tag}`, { verbose })
+  if (action === 'update') {
+    const { data: { name, id: release_id } } = await octokit.request('GET /repos/:owner/:repo/releases/tags/:tag', {
+      owner,
+      repo,
+      tag,
+    })
+    log(`Release ${name} (ID: ${release_id}) found based on tag: ${tag}`, { verbose })
+
+    if (release_id) {
+      await octokit.request('PATCH /repos/:owner/:repo/releases/:release_id', {
+        owner,
+        repo,
+        release_id,
+        body: formatted,
+      })
+      log(`Release ${name} (ID: ${release_id}) updated with notes: ${formatted}`, { verbose })
+    }
+  }
 
-  if (release_id) {
-    await octokit.request('PATCH /repos/:owner/:repo/releases/:release_id', {
+  if (action === 'tag') {
+    const { data: { tag_name } } = await octokit.request('POST /repos/:owner/:repo/releases', {
       owner,
       repo,
-      release_id,
-      body: formatted,
+      tag_name: tag,
+      name: tag,
+      // prerelease: default false
```

Comment:
what could be something inside the commit message that would indicate it is a pre-release? for instance the last part of an alpha version like v3.5.2-alpha-1  ?

Reply:
I think anything beyond a regular semver’s x.y.z scope can be considered prerelease (like what we have in gh actions matching for release, `v*-*` is “alpha”-like, and `v*` without the suffix is a proper release)